### PR TITLE
CAMEL-11231 add the JsonAPI Data Format

### PR DESCRIPTION
**Pull request by CodeSmell**
Updated from earlier PR 
https://github.com/apache/camel/pull/2844

Moved `JsonApiDataFormat` to Camel component
Will add additional Data Format integration w/ a little guidance :)

### CODE REVIEW

File: components/camel-jsonapi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    </dependency>
+    
+    <dependency>
+        <groupId>com.github.jasminb</groupId>
+        <artifactId>jsonapi-converter</artifactId>
+        <!--
+        <version>${jasminb-jsonapi-version}</version>
+        
+        -->
+        <version>0.9</version>
```
Comment (oscerd) [inline components/camel-jsonapi/pom.xml]: This can be a property in parent/pom

**Comment by CodeSmell** [comment]
Let me know if going to merge this, or if I should keep adding the other stuff here

**Comment by oscerd** [comment]
Can you take as example other dataformat to align also the camel-core for this? For adding the component to the kit I'll take care of it

**Comment by CodeSmell** [comment]
That would essentially be adding the DataFormat definition to the model 
I'll "copy" the changes made for the Thrift format

**Comment by CodeSmell** [comment]
I know there is a bit more to do on the `org.apache.camel.model.dataformat.JsonApiDataFormat` but wanted to get the rest out there and see if I was missing anything else

**Comment by CodeSmell** [comment]
There were generated files, wasn't sure if I should add those?

Anything else that needs to be done?

**Comment by oscerd** [comment]
> There were generated files, wasn't sure if I should add those?
> 
> Anything else that needs to be done?

is it the component starter stuff? It must be added.

**Comment by oscerd** [comment]
Let me try to merge this week.

**Comment by oscerd** [comment]
Merged on master. Thanks

**Comment by oscerd** [comment]
We need a karaf feature and integration tests.

**Comment by oscerd** [comment]
Can you please add more docs on src/main/docs/jsonApi-dataformat.adoc? @CodeSmell

**Comment by CodeSmell** [comment]
I'll take a look at this

**Comment by oscerd** [comment]
I added the karaf feature and integration tests meanwhile
thanks for the contribution @CodeSmell !

**Comment by CodeSmell** [comment]
Thanks for the help. Not sure I fully understand how the data format plugs into camel though 

Do you want me to look at the docs? Anything else?

**Comment by oscerd** [comment]
yes, in the docs you may add more examples :-) Thanks!

**Pull request closed without merging**